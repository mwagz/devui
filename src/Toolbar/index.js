export Toolbar from './styles/Toolbar';
export Divider from './styles/Divider';
export Spacer from './styles/Spacer';
