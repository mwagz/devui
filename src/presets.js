/* eslint-disable file */
// import installExtension, {
//   EMBER_INSPECTOR, REACT_DEVELOPER_TOOLS,
//   BACKBONE_DEBUGGER, JQUERY_DEBUGGER,
//   ANGULARJS_BATARANG, VUEJS_DEVTOOLS,
//   REDUX_DEVTOOLS, REACT_PERF,
//   CYCLEJS_DEVTOOL,
// } from 'electron-devtools-installer';
import { reducer } from 'redux-devtools-extension';
import { createStore } from 'redux';
import devToolsEnhancer from 'remote-redux-devtools';
import 'codemirror/mode/javascript/javascript';
import 'codemirror/addon/fold/foldgutter';
import 'codemirror/addon/fold/foldcode';
import 'codemirror/addon/fold/brace-fold';

if (process.env.NODE_ENV !== 'test') {
  require('../fonts/index.css');
  require('codemirror/lib/codemirror.css');
  require('codemirror/addon/fold/foldgutter.css');
}
const store = createStore(reducer);
createStore(store, reducer, devToolsEnhancer());


// const store = createStore(reducer, preloadedState, devToolsEnhancer().then let store => {
//   subscribe({
//      this.store.subscribe(preloadedState, store(( => {
//           dispatch();
//       });
//     })
// }));
