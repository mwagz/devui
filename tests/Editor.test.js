import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import { Editor } from '../src';
import 'codemirror/mode/javascript/javascript';

describe('Editor', function () {
  const getBoundingClientRect = jest.fn();
  const getClientRects = jest.fn();
  document.body.createTextRange = function () {
    return {
      getBoundingClientRect() {
        getBoundingClientRect();
        return {};
      },
      getClientRects() {
        getClientRects();
        return {};
      }
    };
  };
  const wrapper = mount(<Editor value="var a = 1;" />);

  it('renders correctly', () => {
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('calls getBoundingClientRect', () => {
    expect(getBoundingClientRect).toBeCalled();
  });

  it('calls getClientRects', () => {
    expect(getClientRects).toBeCalled();
  });
});
