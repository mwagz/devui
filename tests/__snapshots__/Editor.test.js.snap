exports[`Editor renders correctly 1`] = `
<Editor
  autofocus={false}
  foldGutter={true}
  lineNumbers={true}
  lineWrapping={false}
  mode="javascript"
  readOnly={false}
  value="var a = 1;">
  <styled.div
    innerRef={[Function]}>
    <div
      className="sc-gZMcBi eROHUl" />
  </styled.div>
</Editor>
`;
